core/base/src/TDirectory.cxx

```diff
@@ -96,8 +96,12 @@ TDirectory::TDirectory(const char *name, const char *title, Option_t * /*classna
 
 TDirectory::~TDirectory()
 {
-   if (!gROOT) {
-      delete fList;
+   // Use gROOTLocal to avoid triggering undesired initialization of gROOT.
+   // For example in compiled C++ programs that don't use it directly.
+   if (!ROOT::Internal::gROOTLocal) {
```

Comment:
Not really important, but why 2 nested ifs instead of one single condition?

Reply:
`delete` can be called on a nullptr, so the extra test is redundant.